src/rgw/rgw_admin.cc

```diff
@@ -6701,6 +6701,7 @@ int main(int argc, const char **argv)
         return -EINVAL;
       }
       std::unique_ptr<rgw::sal::RGWRole> role = driver->get_role(role_name, tenant, path, assume_role_doc);
+      role->update_max_session_duration(max_session_duration);
```

Comment:
This will work, but please pass the max_session_duration param in the get_role() call above, something like the following:
`std::unique_ptr<rgw::sal::RGWRole> role = driver->get_role(role_name, tenant, path, assume_role_doc, max_session_duration, {});`
`update_max_session_duration` was meant to update the existing max_session_duration.

Reply:
That's a good point @pritha-srivastava. Added a new evidence.